src/lib/schema.ts

```diff
@@ -25,50 +25,61 @@ interface File {
   url_path: string | null;
   filetype: string | null;
   metadata: MetaData | null;
+  [key: string]: string | null | MetaData | undefined;
 }
 
 class MddbFile {
   static table = Table.Files;
   static supportedExtensions = ["md", "mdx"];
 
-  _id: string;
-  file_path: string;
-  extension: string;
-  url_path: string | null;
-  // TODO there should be a separate table for filetypes
-  // and another one for many-to-many relationship between files and filetypes
-  filetype: string | null;
-  metadata: MetaData | null;
+  file: File = {
+    _id: "",
+    file_path: "",
+    extension: "",
+    url_path: null,
+    filetype: null,
+    metadata: null,
+  };
 
   // TODO type?
   constructor(file: any) {
-    this._id = file._id;
-    this.file_path = file.file_path;
-    this.extension = file.extension;
-    this.url_path = file.url_path;
-    this.filetype = file.filetype;
-    this.metadata = file.metadata ? JSON.parse(file.metadata) : null;
+    this.file._id = file._id;
+    this.file.file_path = file.file_path;
+    this.file.extension = file.extension;
+    this.file.url_path = file.url_path;
+    this.file.filetype = file.filetype;
+    this.file.metadata = file.metadata ? JSON.parse(file.metadata) : null;
+
+    // Assign dynamic properties using index signature to this.file
+    for (const key in file) {
+      if (
+        key !== "_id" &&
+        key !== "file_path" &&
+        key !== "extension" &&
+        key !== "url_path" &&
+        key !== "filetype" &&
+        key !== "metadata"
+      ) {
+        this.file[key] = file[key];
+      }
+    }
   }
 
   toObject(): File {
-    return {
-      _id: this._id,
-      file_path: this.file_path,
-      extension: this.extension,
-      url_path: this.url_path,
-      filetype: this.filetype,
-      metadata: this.metadata,
-    };
+    return { ...this.file };
   }
 
-  static async createTable(db: Knex) {
+  static async createTable(db: Knex, properties: string[]) {
     const creator = (table: Knex.TableBuilder) => {
       table.string("_id").primary();
-      table.string("file_path").unique().notNullable(); // Path relative to process.cwd()
-      table.string("extension").notNullable(); // File extension
-      table.string("url_path"); // Sluggfied path relative to content folder
-      table.string("filetype"); // Type field in frontmatter if it exists
-      table.string("metadata"); // All frontmatter data
+      table.string("file_path").unique().notNullable();
+      table.string("extension").notNullable();
+      table.string("url_path");
+      table.string("filetype");
+      table.string("metadata");
+      for (let index = 0; index < properties.length; index++) {
+        table.string(properties[index]);
+      }
```

Comment:
What happens if a computed property returns an array or a boolean? Shouldn't it be `table.<some type>`?  Also, were the comments incorrect?

```suggestion
      properties.forEach((property) => {
        table.string(property);
      })
```

Reply:
I see what you are saying, I thought about a couple of scenarios:
- a field is array => there's no way to store arrays in SQL
- a field is a number => what type of number? float, double, ...
- what if a field is an object => there's no way to store objects in SQL
- what if a field has a different type ( number/string ) across two files, should it be defaulted to string?

Reply:
Well, use any SQL types that are most suited for each JS/TS type. But it's not like everything needs to be stringified 😅 If it's an object/array, stringify to JSON. 

> what if a field has a different type ( number/string ) across two files, should it be defaulted to string?
You could type the computed function signature in a way that it always should return a single type defined by the user. But for now, ok, just stringify everything. We'll properly implement it when we move computed fields definitions to their final destination = document types.